app/src/main/java/io/appium/settings/LocationTracker.java

```diff
@@ -244,20 +249,24 @@ public synchronized Location getLocation(Context context) {
 
         if (isPlayServicesConnected()) {
             try {
-                //noinspection deprecation
                 mLocation = LocationServices.FusedLocationApi.getLastLocation(mGoogleApiClient);
                 if (mLocation != null) {
                     // Make sure the fallback is removed after play services connection succeeds
                     if (isLocationManagerConnected()) {
                         stopLocationUpdatesWithoutPlayServices();
                     }
                     return mLocation;
+                } else {
+                    stopLocationUpdatesWithPlayServices();
```

Comment:
should we always stop google services if the returned location is null? I mean it still might need some time to receive the actual location after being connected

Reply:
I was thinking about that. I guess there's no harm having both services running.